effort_controllers/src/joint_group_effort_controller.cpp

```diff
@@ -40,8 +40,10 @@ JointGroupEffortController::init(
   }
 
   try {
-    // undeclare interface parameter used in the general forward_command_controller
-    get_node()->undeclare_parameter("interface_name");
+    // Explicitly set the interface parameter declared by the forward_command_controller
+    // to match the value set in the JointGroupEffortController constructor.
+    get_node()->set_parameter(rclcpp::Parameter("interface_name", hardware_interface::HW_IF_EFFORT));
+
   } catch (const std::exception & e) {
```

Comment:
I would use this opportunity also to make this catch more specific. According to docstring `set_parameter` method throws `rclcpp::exceptions::ParameterNotDeclaredException`. Should be make this more explicit? @schornakj, @bmagyar what do you think?

Reply:
I think I'd actually continue catching `std::exception`. If a different exception is thrown other than `ParameterNotDeclaredException` (which is not likely) we still want to catch it and cleanly exit the initialization function by returning an error code.